app/models/invoice.rb

```diff
@@ -24,6 +24,10 @@ class ReadOnlyError < StandardError; end
 
   validates :date, uniqueness: { scope: :partner_id }
 
+  # Ensure these two values are numbers even though field is a string
+  validates :amount,  numericality: true, allow_nil: true
+  validates :finalized_amount,  numericality: true, allow_nil: true
```

Comment:
Maybe we can even validate further with one of these

 https://guides.rubyonrails.org/active_record_validations.html#numericality

Perhaps ```:greater_than_or_equal_to: 0```

Not sure if that would clash with nil though, totally your call, just food for thought 🍕

Reply:
Unfortunately we can't allow_nil if we do this 😢